src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -50,19 +53,39 @@ class SDCPUThread
     class SDCPUExecContext : public ExecContext
     {
       protected:
-        SDCPUThread& backingState;
+        InflightInst& backingState;
         
       public:
         SDCPUExecContext(SDCPUThread& backing_state):
             backingState(backing_state)
         { }
     };
 
-    class DynamicInst
+    class InflightInst
     {
-        StaticInst& instructionRef;
-        // TODO instruction tag
-        // TODO instruction results
+      public:
+        enum Status
+        {
+            Invalid = 0,
+            Waiting, // Can't be executed because dependencies remain
+            Executing, // Requirements met, but waiting for results
+            Completed // Results have been received, but not yet committed.
+        };
+
+        // Where am I in completing this instruction?
+        Status status;
+
+        // What is this instruction I'm executing?
+        InstSeqNum tag;
+        StaticInstPtr instRef;
+
+        // How does this instruction influence my system?
+        vector<InstResult> results;
+        vector<TheISA::CCReg> ccResults;
+        Fault fault;
+
+        // TODO deal with macroops?
```

Comment:
I believe that `InflightInst` will be micro-ops (or just ops) and that the macro-ops will be broken down in decode before the `InflightInst` is created. In fact, you may want to add an `assert` or `panic` in the constructor to check this.